Aperture Orsay driver

Kleijwegt opened the pull request.

Commit:
Composited detector and dataflow for use of FIB/SEM

Commit:
Alignment tab Enzel for 3 beam alignment, single frame acquisition and simulation files Enzel (with the configuration as the Hw based startup file)

pieleric committed:
doc: Add METEOR roles and update ENZEL

Latest way it's implemented.

pieleric committed:
acq align coordinates: direclty use .max() and .min()

Just small code simplification, to make it clearer.

pieleric committed:
gui: add Z localization widgets

To select the stigmator angle, start a localization, and report the
result to the user.

Also add the actual procedure as a Future, in odemis.acq.align.z_localization.

pieleric committed:
gui model: automatically select the new CryoFeature when created

It's odd that the feature is created but not selected.

pieleric committed:
gui cont: move cryo acquired streams control to the streambar controller

Simplifies the tab controller which was doing a lot of different things.

Now there is a dedicated streambar controller for Cryo localization.
In addition to hiding/showing the streams when the view is selected,
it does the same when a feature is selected.

Among other things, it fixes a bug where *every* stream was hidden when changing feature.

Commit:
FASTEM: use calibration manager also during acquisition

When acquiring an ROA we run a set of calibrations before each ROA. So far there was own code for this in the acquisition manager. However, now that we have the calibration manager, make use of it.

Commit:
FASTEM: some clean-ups and improved logging

Commit:
progressive batch future: fix start time in progress update (#2105)

* progressive batch future: fix start time in progress update

When the progress was set the start time was not correctly updated.
If the sub-future was cancelled, also cancel the batch future.
Add test cases for batch future.

Commit:
FASTEM: update estimated acquisition time when dwell time is adjusted

Commit:
FASTEM: update estimated acquisition time when resizing ROA

tepals committed:
Fastem acquisition manager: add and update logging to have more info during debugging.

Commit:
Only skip the arpolarimetry test if the module is not found

Commit:
Fixed the enzel test case by updating the range of the sim file. The range is increased as on the real hardware startup file.

Commit:
Updated naming of TestSpectrograph --> SpectrographTestBaseClass to prevent testing of the base class

Commit:
Fixing a comparison between a string and an image --> check if key is in metadata

tepals committed:
Github workflow: added test cases from dataio and acq.drift and removed tests from model because they kept failing.

pieleric committed:
gui test: remove unused sandbox.py

A file which we had used for tests, a long time ago. Never used
directly.

Commit:
FASTEM: bug fix acquisition (don't pass detector rotator anymore in caller)

Bug introduced in commit 43b86fe85ffb41b48a2937b7abccb8d225dc7f17

Commit:
FASTEM: call calibration module using Enum constant

Commit:
Merge pull request #2115 from Sabrina85/fastem-bug-fix-calib-manager-in-acq

FASTEM: bug fix acquisition (don't pass detector rotator anymore in caller)

pieleric committed:
Merge pull request #2114 from pieleric/gui-test-remove-unused-sandbox-py

gui test: remove unused sandbox.py

Commit:
FASTEM: some improvements to the calibration manager run method

Commit:
Merge pull request #2116 from Sabrina85/fastem-enum-calib

FASTEM: call calibration module using Enum constant

pieleric committed:
Merge pull request #2112 from Kleijwegt/fixing_test_cases

Fixing test cases, 4 small commits for different failing test cases.

tepals committed:
Acq align: remove from .fastem import align from __init__.py to prevent circular import.

In odemis.acq.align.fastem the fastem calibrations module was imported which eventually led to the import: from odemis.acq.align import FindEbeamCenter.
Removeing from .fastem import align removed the circular import.

pieleric committed:
gui cont tabs: allows to select multiple files to open in analysis tab

It's possible to open multiple files simultaneously, but until now the user had to
select each file one at a time using the "From file..." button.
Now, it's directly possible to select multiple files. Either from the
"open" button, or from the "From file...".

Handy when trying to open quite a few files at the same time, like
tiles.

pieleric committed:
Merge pull request #2093 from pieleric/gui-cont-refactor-cryo-acquire-stream-controller

gui cont: refactor cryo acquire stream controller

Commit:
Aperture driver implementation

pieleric committed:
gui comp stream: remove "improvement" to event handler which broke closing streams

In commit a8fec30dee4 (move cryo acquired streams control to the streambar
controller), I tried to "clean" the event removal. However, as the
control is destroyed, the event is automatically removed. In some cases,
this would even cause an error as we were accessing dead objects.

pieleric committed:
acq drift test: improve reliability

Force the "randomness" to always be the same one so that the tests
always pass or fail, independent of when you're running them.

Commit:
Merge pull request #2117 from Sabrina85/fastem-some-improvements-to-the-calibration-manager-run-method

FASTEM: some improvements to the calibration manager run method

Commit:
crosshair: increase thickness crosshair to increase visibility (#2106)

* crosshair: increase thickness crosshair to increase visibility

Especially on an image the crosshair is not that clearly visible. Increase the thickness by 1 pixel to increase visibility.

pieleric committed:
Merge pull request #2124 from pieleric/acq-drift-test-improve-reliability

acq drift test: improve reliability

pieleric committed:
driver andorcam3: fix incorrect warning

Missing break in for loop caused every gain to be reported as "unhandled",
although everything worked fine.

tepals committed:
util conversion: wave2rgb refactored to become more readable and reduce the number of lines.

wave2rgb renamed to wavelength2rgb

pieleric committed:
util units: simplify round_down_significant and improve docstring

From code golf session (only kepts the non-crazy part)

Commit:
Merge pull request #1963 from Kleijwegt/3DOF_GUI_squash

3 Beam alignment Enzel

Commit:
Merge pull request #2125 from Sabrina85/fastem-some-clean-ups-and-improved-logging

* FASTEM: some clean-ups and improved logging

Commit:
fastem: run all calibrations

Now run all calibrations, but in the calibration module we make sure to only log the values for now.

Commit:
bug fix in logs

Commit:
Merge pull request #2127 from Sabrina85/fastem-run-all-calibrations

fastem: run all calibrations

tepals committed:
driver technolution: PEP8 clean up

tepals committed:
driver technolution: Send custom metadata to the ASM during fastem acquisition.

The ASM accepts a custom data field to add any metadata. It can be useful to know with what exact settings a megafield was acquired, therefore add calibration data as custom metadata.
This will be stored in the mega_field_meta_data.yml file that is stored with each megafield acquisition.
added test cases

Commit:
fastem calibrations: make order same as in orginal run script

The descan gain and the scan amplitude pre alignment were in the original main run script in opposite order. In principle these two calibrations are independent of each other. However, to avoid confusion, make it the same order in Odemis.

Commit:
Merge pull request #2128 from Sabrina85/fastem-reorder-calib

fastem calibrations: make order same as in orginal run script

pieleric committed:
Merge pull request #2123 from pieleric/gui-comp-stream-remove-improvement-to-event-handler-which-broke-closing-streams

gui comp stream: remove "improvement" to event handler which broke closing streams

pieleric committed:
Merge pull request #2129 from pieleric/driver-andorcam3-fix-incorrect-warning

driver andorcam3: fix incorrect warning

pieleric committed:
plugins ext_spec_map: new plugin for spectrum acquisition controlled via external e-beam

pieleric committed:
gui util widgets: fix error when disconnecting a AxisConnector

Probably we never tried (except when stopping the GUI).
Need to set the argument name explicitly, as for the VAConnector.

pieleric committed:
gui cont streams: fix display order of HW Detector VAs

HW VAs are mixed for the emitter and detector, but until now we only
used the emitter order. That meant that if the VAs came from the
detector, the order would be roughly random (except for the VAs who
share the same name).
=> use both orders, one after the other. It's not an issue if a VA is
listed several times, the first occurence will be used.

Commit:
FASTEM: Return resolution and cell size for cell translation calibration

The cell translation calibration is run for a specific set of values for the single field resolution and the cell complete resolution. Thus, for the acquisition it is necessary to also use this settings. As long as adjustable resolution and cell size are not supported in the calibration itself, also do not allow to use different value for the acquisition.

Commit:
Merge pull request #2132 from Sabrina85/fastem-cell-translation-return-value

FASTEM: Return resolution and cell size for cell translation calibration

tepals committed:
driver xt_client: allow to set the brightness and contrast (#2118)

* driver xt_client: allow to set the brightness and contrast

For the overview imaging it is necessary to adjust the brightness and contrast beforehand.
Add a brightness and a contrast VA to allow setting these values.

Move (auto) contrast brightness functions from scanner to detector.

Remove the detector argument from applyAutoFocus and applyAutoContrastBrightness because it was unused.

* Update after review
- set back detector in applyAutoFocus
- fixes in test cases

* Add test case that tests if the brightness and contrast on the hardware actually change and correct docstring.

Commit:
during testing improvements on real Hw

pieleric committed:
gui conf: fallback to default project dir if impossible to create a new project folder

It can happen that the default project dir cannot be created, for
instance, it got unmounted, or permissions changed. Let's not completely
fail to start the GUI in such case.
=> fall back to the default project directory.

Also, add a "/" after the root project dir because it's actually supposed to
contain the *previous* project directory, and then the parent folder is
used. As it's directly the parent folder, we need to indicate that with
an extra "/".

pieleric committed:
dataio hdf5: read/write MD_HW_NAME and MD_HW_VERSION

It's quite handy metadata, to know which camera was used for imaging.
Of course we also store this info in MD_EXTRA_SETTINGS, but on systems
with multiple cameras, you have to guess which camera was used.

pieleric committed:
Merge pull request #2120 from pieleric/gui-cont-tabs-allows-to-select-multiple-files-to-open-in-analysis-tab

gui cont tabs: allows to select multiple files to open in analysis tab

pieleric committed:
util units: add also µsec display to the time display

Avoids having values < 1ms represented as "0 s". The µs can still matter
a lot, depending on which time frame we care about.

pieleric committed:
gui conf data: hide framePeriod and dropOldFrame VAs

New advanced VAs on the andorcam2 driver. No need to let the user care
about them

pieleric committed:
gui comp: increase the text fields in height to ensure all characters are shown

By default the text fields are too small, at least with the default
Ubuntu theme. The bottom of letters like p or µ was hidden.

So reduce the margin and increase the text height. It
should actually take a few pixels *less* in total now.

pieleric committed:
driver andorcam2: fix minimum resolution now accepted

The cameras may have a minimum of pixels to send (more than 1). For very
tiny resolutions in theory this could be a limit. Until now we
mistakenly applied this minimum both for X and Y. This prevented some
cameras to use full vertical binning, and instead generated very unclear
errors (empty AssertionError).

=> Correctly check the size and, in the very rare case it's actually
limiting, try to increase a bit the resolution.

tepals committed:
Driver tfsbc: added logging to get information about which dc coil is out of range.

pieleric committed:
Merge pull request #2138 from pieleric/acq-align-coordinates-direclty-use-max-and-min

acq align coordinates: direclty use .max() and .min()

pieleric committed:
Merge pull request #2121 from pieleric/doc-add-meteor-roles-and-update-enzel

doc: Add METEOR roles and update ENZEL

pieleric committed:
model dataflow: introduce HwTrigger Event

New Event type to indicate that the event will from from an external,
hardware input.

Will be used by the andorcam2 driver.

pieleric committed:
driver andorcam2: handle hw trigger

Add support for hardware trigger, by providing a special .hardwareTrigger Event.
When the DataFlow is synchronized with this Event, it requests the
camera to use the hardware trigger for frame start.
Provide a hw_trigger_invert argument to control the trigger inversion if
needed.

Also add discardOldFrames and framePeriod for extra control.

pieleric committed:
Merge pull request #2126 from tepals/refactor-wave2rgb

util conversion: wave2rgb refactored to become more readable and redu…

pieleric committed:
Merge pull request #2136 from pieleric/dataio-hdf5-read-write-md_hw_name-and-md_hw_version

dataio hdf5: read/write MD_HW_NAME and MD_HW_VERSION

pieleric committed:
Merge pull request #2135 from pieleric/gui-conf-fallback-to-default-project-dir-if-impossible-to-create-a-new-project-folder

gui conf: fallback to default project dir if impossible to create a new project folder

Commit:
update the log filter of the test cases to display the start and end part

Commit:
added a test case for filter_test_log

tepals committed:
util test: stop_backend updated to call odemis-stop when it fails to stop the odemis backend.

pieleric committed:
Merge pull request #2130 from pieleric/util-units-simplify-round_down_significant-and-improve-docstring

util units: simplify round_down_significant and improve docstring

Commit:
Merge pull request #2144 from tepals/force-stop-test-stop-backend

util test: stop_backend updated to call odemis-stop when it fails to …

Commit:
Merge pull request #2143 from Kleijwegt/log_filter_change

update the log filter of the test cases to display the start and end part

pieleric committed:
config: update debian rules to use pybuild

Also clean up Cython C files.

Commit:
Updated the image for the Odemis-viewer installer. Includes both the new logo and a more relevant image.

pieleric committed:
Merge pull request #2145 from Kleijwegt/installer_update

Updated the image for the Odemis-viewer installer. Includes both the new logo and a more relevant image.

tepals committed:
driver xt_client: prepareForScan was only added to the FIBScanner and not to the normal Scanner.

This caused the acquisition code in the detector to fail. This commit adds the prepareForScan to the Scanner class, and moves beam blanking to that function, because the FIBScanner does not support beam blanking.

There were also a few test cases which overwrote the blanker VA with a boolean value, which is also fixed.

pieleric committed:
config: remove executable bit from plugins

They are not made to be run directly from the terminal, so it doesn't
make sense to have an executable bit. They should only be loaded as
extra modules by the GUI.

pieleric committed:
Merge pull request #2122 from pieleric/gui-add-z-localization-widgets

gui: add Z localization widgets

Commit:
Supress the warning that numpy.log10 divides by zero in _projection.py

pieleric committed:
Merge pull request #2133 from pieleric/gui-cont-streams-fix-display-order-of-hw-detector-vas

gui cont streams: fix display order of HW Detector VAs

pieleric committed:
Merge pull request #2134 from pieleric/gui-util-widgets-fix-error-when-disconnecting-a-axisconnector

gui util widgets: fix error when disconnecting a AxisConnector

pieleric committed:
acq stream helper: don't show integrationTime also if HW exposureTime is used

To avoid confusion, if the stream explicitly shows the exposureTime VA,
we never provide an integrationTime. However, there are 2 ways link the
exposureTime to the stream. So need to check both ways.

Matters for the ext_spec_map plugin, which uses exposureTime as a HW VA.

tepals committed:
dataio catmaid: skip unittests as the external servers are often down, and we do not use catmaid yet.

pieleric committed:
driver andorcam2: handle change of sync type during acquisition

It's very unlikely, but in theory, it's allowed to change the
synchronization type (eg, None -> SW trigger -> HW trigger) while
the acquisition is running. So far, this was not handled properly.
In particular, if synchronized on HW trigger, and switch to software
trigger, or nothing it'd still wait for the trigger, although it
should start a new acquisition again.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
             self._executor = None
+
+
+NMBR_APERTURES = 30
```
It seems to me that you can use `SelectedDiaph.Max` instead of hard-coding it.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+                "Size": model.FloatVA(0.000, readonly=True),
+                "PositionX": model.FloatVA(0.000, ),
+                "PositionY": model.FloatVA(0.000, )
+            }
```
We cannot have VAs in metadata. The metadata should contain just simple data types. I'd suggest to not use the "OrsayParameterConnector", and just have dedicated functions that listen to the parameters and update the metadata whenever it updates. As they all should be read-only, it should be relatively easy.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        for aprtr_nmbr in range(NMBR_APERTURES):
+            self._apertureConnectors[
+                f"Aperture{aprtr_nmbr}"] = []  # Connectors are saved for refs, so order doesn't matter
+            self._apertureDict[f"Aperture{aprtr_nmbr}"] = {
```
As a key, let's just use the aperture number (int). It'll be easier to link the "selectedApeture" to the metadata.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        self._metadata[model.MD_APERTURES_INFO] = self._getAllApertureMetadata()
+
+    def selectAperture(self, id):
+        return self._executor.submit(self._selectAperture, id)
```
As discussed previously, let's not have a special method.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+NMBR_APERTURES = 30
+
+
+class FIBAperture(model.Actuator):
```
A docstring would be welcome here, especially to indicate the expected usage with respect to the position X/Y and selected aperture. Don't hesitate to have a long description. He is the best place to indicate 1) what this class is supposed to help for, and 2) how to use it (intended usage).

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        """
+        reference usually takes axes as an argument. However, the SmarPod references all
+        axes together so this argument is extraneous.
+        """
```
Unfortunately the SmarPod driver was incorrect (and not corrected because we didn't use it). `.reference()` *must* take one argument `axes`. Please add it, and call ` self._checkReference(axes)` to check it only contains allowed axes. In the docstring, you can mention that in any case, all axes will be referenced.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        # TODO K.K. check for alternative stopping options, speed and actions if this doesn't work
+        # Wait for max 2 seconds for the movement to stop
+        while not self._hybridAperture.IsMovingX.Actual == "False" and not self._hybridAperture.IsMovingY.Actual == "False" \
+                or t + 1 > time.time():
```
It says "max 2 seconds", but the code has a `+ 1`. Please update. Possibly, you could just write `tend = time.time() + 1`, and then use `tend` in the test.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        while not self._hybridAperture.IsMovingX.Actual == "False" and not self._hybridAperture.IsMovingY.Actual == "False" \
+                or t + 1 > time.time():
+            time.sleep(0.01)
+        if not self._hybridAperture.IsMovingX.Actual and not self._hybridAperture.IsMovingY.Actual:
```
`isMovingX/Y.Actual` are strings containing "True" or "False". So their value will always be considered `True`. So this test will always be `False`,
You could rewrite the whole block like that:
```python
while time.time() < tend:
     if  self._hybridAperture.IsMovingX.Actual == "False" and self._hybridAperture.IsMovingY.Actual == "False":
        break
     time.sleep(0.01)
else:
    # Timeout
   : the code from here
```

pieleric commented on src/odemis/model/_metadata.py:
```diff
@@ -58,6 +58,7 @@
 MD_LENS_NAME = "Lens name"  # str, product name of the lens
 MD_LENS_MAG = "Lens magnification"  # float (ratio), magnification factor
 MD_LENS_NA = "Lens numerical aperture"  # float (ratio), numerical aperture
+MD_APERTURES_INFO = "Information about all the apertures in the system" # dict of dicts with lifetime, size, positionX and position Y as keys.
```
Let's move this line to near MD_CALIB, which is more related.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+            msg = "The aperture moved is not properly canceled the current status of x is %s and of y is %s" % \
+                  (self._hybridAperture.IsMovingX.Actual, self._hybridAperture.IsMovingY.Actual)
+            logging.debug("Forcing the Actual value of IsMovingX and IsMovingY to false")
+            self._hybridAperture.IsMovingX.Actual = False
+            self._hybridAperture.IsMovingY.Actual = False
```
Does that really work?! I believe it would work on the simulator, but doubt it works on the real hardware.

Kleijwegt replied:
Looking again at the code I'm also wondering that. I added a comment and I will test that next time.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+            logging.debug("Forcing the Actual value of IsMovingX and IsMovingY to false")
+            self._hybridAperture.IsMovingX.Actual = False
+            self._hybridAperture.IsMovingY.Actual = False
+            raise HwError(msg)
```
I think a TimeoutError would be more appropriate here. Normally we use HwError to report an error that the hardware expresses (or when we know that the hardware is down/disconnected).

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        tend = time.time() + timeout
+        self._hybridAperture.Calibrated.Target = True
+        while self._hybridAperture.Calibrating.Actual == True \
+                or self._hybridAperture.Calibrated != True:
```
`self._hybridAperture.Calibrated` -> `self._hybridAperture.Calibrated.Target`

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        self._hybridAperture.YPosition.Target = pos["y"]
+
+        # TODO K.K. test and check isMovingX/Y with Wilco
+        while not float(self._hybridAperture.XPosition.Actual) != pos \
+                or not float(self._hybridAperture.YPosition.Actual) != pos \
```
When you compare floats, especially when converted from string, you have to use `almost_equal`. Here, you might even want to "bump up" the absolute error to a tiny value (eg 1µm), to make sure that if the position is pretty close, that counts.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+        logging.debug(f"Aperture {id} is selected")
+        tend = time.time() + timeout
+        self._hybridAperture.SelectedDiaph.Target = id
+        while int(self._hybridAperture.SelectedDiaph.Actual) != id \
+                and self._hybridAperture.IsMovingX.Actual == "True" \
+                and self._hybridAperture.IsMovingY.Actual == "True":
```
Why use a `and` here? As soon as one axis stops, it'd be considered completed. I think you should use `or`, like in `_doMoveAbs()`.

Kleijwegt replied:
I removed this function. Indeed and is wrong here.

pieleric commented on src/odemis/driver/orsay.py:
```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+class FIBAperture(model.Actuator):
+    def __init__(self, name, role, parent, **kwargs):
+        # TODO K.K. define a more sane range for the ranges
+        axes_def = {"x": model.Axis(unit="m", range=(-6e-3, 6e-3)),
+                    "y": model.Axis(unit="m", range=(-6e-3, 6e-3))}
```
Can you get the ranges from the .Max and .Min of the orsay parameters maybe? Otherwise, put them in a constant.

pieleric committed:
Merge pull request #2140 from pieleric/util-units-add-also-µsec-display-to-the-time-display

util units: add also µsec display to the time display

pieleric committed:
Merge pull request #2149 from tepals/catmaid-skip-unittests

dataio catmaid: skip unittests as the external servers are often down…

pieleric committed:
Merge pull request #2151 from pieleric/gui-stream-hide-some-non-useful-va-from-andorcam2

gui stream: hide some non useful VA from Andorcam2

pieleric committed:
Merge pull request #2137 from pieleric/gui-comp-increase-the-text-fields-in-height-to-ensure-all-characters-are-shown

gui comp: increase the text fields in height to ensure all characters are shown

pieleric committed:
Merge pull request #2150 from pieleric/config-minor-improvements-for-debian-build

config: minor improvements for debian build

pieleric committed:
Merge pull request #2142 from pieleric/plugins-ext_spec_map-new-plugin-for-spectrum-acquisition-controlled-via-external-e-beam

plugins ext_spec_map: new plugin for spectrum acquisition controlled via external e-beam

Commit:
Merge pull request #2147 from Kleijwegt/supress_overflow_warning

Supress the warning that numpy.log10 divides by zero in _projection.py

pieleric committed:
gui util img: add label of each metadata in legend

When exporting an image, some metadata in the legend was written like
"Exposure time: 1s" and other like "15 kV". The latter version was sort
of understandable, but not very much, and the mix was odd.
=> Put a label for all metadata.

pieleric committed:
config: fix secom-pretty sim to match the images again

Some time ago the CCD simulator must have adjusted the metadata, and
this caused the images to not overlay properly anymore.

It's now easier, we just put the original lens magnification.

pieleric committed:
driver andorcam2: fix docstring

Commit:
Processing review Eric

pieleric committed:
Merge pull request #2152 from pieleric/gui-util-img-add-label-of-each-metadata-in-legend

gui util img: add label of each metadata in legend

Commit:
Merge pull request #2108 from Kleijwegt/orsay_adjust_new_API

Getting the orsay test cases to pass again

pieleric committed:
Merge pull request #2153 from pieleric/config-fix-secom-pretty-sim-to-match-the-images-again

config: fix secom-pretty sim to match the images again

pieleric committed:
Merge pull request #2141 from pieleric/driver-andorcam2-add-hardware-trigger-support

driver andorcam2: add hardware trigger support

pieleric committed:
gui model: fix error on ENZEL/METEOR if there is stigmator without MD_CALIB

The code was supposed to handle it... but missed a "return" so it would
detect the MD_CALIB is missing, and then fail anyway.

pieleric committed:
gui model: raise an clear error if ENZEL has no FIB components

With the new alignment tab, the FIB is required.
=> Properly report the components are missing in the YAML file

Also add them to the standard ENZEL simulator. It doesn't yet work
perfectly because the SimSEM has a .shift VA instead of .beamShift VA on
the XT. Probably the former name should be used everywhere.

Take the opportunity to rename the special XT ENZEL sim to a simpler name.

Commit:
Merge pull request #2157 from pieleric/gui-fix-minor-errors-with-enzel

gui: fix minor errors with ENZEL

Kleijwegt commented:
Due to rebasing and having a PR on a PR this one become unreadable
